aui.views/src/AUI/Platform/linux/ADBus.cpp

```diff
@@ -111,28 +111,76 @@ DBusHandlerResult ADBus::listener(DBusConnection     *connection,
 }
 void ADBus::deleter(void* userdata) noexcept {
     auto listener = reinterpret_cast<RawMessageListener*>(userdata);
-    auto listenerIt = std::find_if(listener->parent->mListeners.begin(),
-                                   listener->parent->mListeners.end(),
-                                   [&](const RawMessageListener& lhs) {
-        return &lhs == userdata;
-    });
+    auto listenerIt = std::find_if(
+        listener->parent->mListeners.begin(), listener->parent->mListeners.end(),
+        [&](const RawMessageListener& lhs) { return &lhs == userdata; });
 
     if (listenerIt == listener->parent->mListeners.end()) {
         return;
     }
     listener->parent->mListeners.erase(listenerIt);
 }
 
-void ADBus::addListener(RawMessageListener::Callback listener) {
-    auto it = mListeners.insert(mListeners.end(), { this, std::move(listener)});
+std::function<void()> ADBus::addListener(RawMessageListener::Callback listener) {
```

Comment:
I think this function is a subject to refactor later. For now, let's keep this as is.

Reply:
#389